src/test/java/racingcar/service/RacingCarServiceTest.java

```diff
@@ -1,53 +1,75 @@
 
-    private static final long gameId = 1L;
+    private static final String TEST_PLAYER_NAME_1 = "hong";
+    private static final String TEST_PLAYER_NAME_2 = "coda";
 
     @Mock
-    private PlayResultDao playResultDao;
-    @Mock
-    private RacingCarDao racingCarDao;
+    private JdbcRacingGameRepository jdbcRacingGameRepository;
     @InjectMocks
     private RacingCarService racingCarService;
 
+    @DisplayName("실제 자동차 게임 실행후 디비에 잘 저장하는지 테스트")
     @Test
     public void saveTest() {
-        final List<Car> carList = List.of(new Car("a", 1), new Car("b", 2));
-        final int tryCountValue = 5;
-        final Cars cars = new Cars(carList);
-        final TryCount tryCount = new TryCount(tryCountValue);
+        final List<RacingCarDto> racingCarDtos = List.of(new RacingCarDto(TEST_PLAYER_NAME_1, 1)
+                , new RacingCarDto(TEST_PLAYER_NAME_2, 2));
+        final RacingCarGameResultDto racingCarGameResultDto = new RacingCarGameResultDto(List.of(TEST_PLAYER_NAME_1),
+                racingCarDtos);
+
+        doNothing().when(jdbcRacingGameRepository)
+                .saveGameResult(anyList(), anyList());
+
+        racingCarService.saveGameResult(racingCarGameResultDto);
+
+        verify(jdbcRacingGameRepository)
+                .saveGameResult(List.of(TEST_PLAYER_NAME_1), racingCarDtos);
+    }
+
+    @DisplayName("게임 실행하고, 반환되는 결과가 예상대로인지 확인하는 기능 테스트")
+    @Test
+    public void playRoundTest() {
+        final List<String> playerNames = List.of(TEST_PLAYER_NAME_1, TEST_PLAYER_NAME_2);
+        final int tryCount = 2;
+        final NumberPicker numberPicker = new DefinedNumberPicker(List.of(4, 4, 6, 2));
 
-        when(playResultDao.insertWithKeyHolder(anyInt(), anyList()))
-                .thenReturn(gameId);
-        doNothing().when(racingCarDao)
-                .insert(anyLong(), anyString(), anyInt());
+        final RacingCarService racingCarService =
+                new RacingCarService(mock(JdbcRacingGameRepository.class), numberPicker);
 
-        racingCarService.saveGameResult(cars, tryCount);
+        final RacingCarGameResultDto racingCarGameResultDto = racingCarService.playRound(playerNames, tryCount);
+        final List<RacingCarDto> racingCarDtos = racingCarGameResultDto.getRacingCars();
 
-        verify(playResultDao).insertWithKeyHolder(tryCount.getValue(), cars.getWinner());
-        verify(racingCarDao).insert(gameId, "a", 1);
-        verify(racingCarDao).insert(gameId, "b", 2);
+        assertAll(
+                () -> assertThat(racingCarGameResultDto.getWinners())
+                        .containsExactly(TEST_PLAYER_NAME_1),
+                () -> assertThat(racingCarDtos)
+                        .extracting(RacingCarDto::getName)
+                        .containsExactly(TEST_PLAYER_NAME_1, TEST_PLAYER_NAME_2),
+                () -> assertThat(racingCarDtos)
+                        .extracting(RacingCarDto::getPosition)
+                        .contains(1, 2)
+        );
```

Comment:
```suggestion
        assertAll(
                () -> assertThat(racingCarGameResultDto.getWinners())
                        .containsExactly(TEST_PLAYER_NAME_1),
                () -> assertThat(racingCarDtos)
                        .extracting("name", "position")
                        .contains(tuple(TEST_PLAYER_NAME_1, 2), tuple(TEST_PLAYER_NAME_2, 1))
        );
```

요렇게 하면 더 간결하게 조합까지 테스트할 수 있겠네요 !

Reply:
이것도 몰랐던 부분이었는데, 감사합니다!!